### PR TITLE
build: Update lower bound of pyhf to v0.7.0

### PR DESCRIPTION
**Pull request by matthewfeickert**
Following the release of `pyhf` `v0.7.0`, update to the latest API and update the lower bounds.

```
* Update lower bound of pyhf to v0.7.0
* Update all usage of `.par_names()` to `par_names`
```

### CODE REVIEW

**Comment by matthewfeickert** [comment]
I can fix this later (unless Alex beats me to it).

**Comment by alexander-held** [comment]
Thanks! #366 is needed first for CI to pass, I'll get that resolved and then we can get this in too.

File: setup.cfg [hunk 1]
```diff
@@ -23,7 +23,7 @@ packages = find:
 package_dir = =src
 python_requires = >=3.7
 install_requires =
-    pyhf[minuit]~=0.7.0rc4  # model.config.suggested_fixed API change, staterror fix, set_poi(None)
+    pyhf[minuit]>=0.7.0  # model.config.suggested_fixed API change, staterror fix, set_poi(None)
```
Comment (alexander-held) [inline setup.cfg:26]: Minor releases of `pyhf` have been breaking `cabinetry` functionality consistently, so I would generally like to stay with the compatible release syntax here (unless you expect that the `pyhf` API is going to be significantly more stable going forward?).

File: src/cabinetry/model_utils.py [hunk 2]
```diff
@@ -642,7 +642,7 @@ def match_fit_results(model: pyhf.pdf.Model, fit_results: FitResults) -> FitResu
 
     bestfit = asimov_parameters(model)  # Asimov parameter values for target model
     uncertainty = prefit_uncertainties(model)  # pre-fit uncertainties for target model
-    labels = model.config.par_names()  # labels for target model
+    labels = model.config.par_names  # labels for target model
```
Comment (alexander-held) [inline src/cabinetry/model_utils.py:645]: This also requires a corresponding change in the test `test_match_fit_results`:
```diff
- mock_model.config.par_names = ["par_c", "par_d", "par_b"]
+ mock_model.config.par_names.return_value = ["par_c", "par_d", "par_b"]
```
There are three instances of that.

File: setup.cfg (hunk 1 above)
Comment (matthewfeickert) [reply setup.cfg:26]: Ah yeah, that was just force of habit. No, I agree that using the compatible release syntax of `pyhf[minuit]~=0.7.0` is a good idea.

**Comment by alexander-held** [comment]
There is also an unrelated new issue in the 3.7 CI (which should work in principle):
```
Theme error:
An error happened in rendering the page advanced.
Reason: ValueError('too many values to unpack (expected 3)')
```
I'll have a look at that.

**Comment by alexander-held** [comment]
For completeness, the related API change for `.par_names` is https://github.com/scikit-hep/pyhf/pull/2027.

File: setup.cfg [hunk 3]
```diff
@@ -23,7 +23,7 @@ packages = find:
 package_dir = =src
 python_requires = >=3.7
 install_requires =
-    pyhf[minuit]==0.7.0rc4  # model.config.suggested_fixed API change, staterror fix, set_poi(None)
+    pyhf[minuit]~=0.7.0  # model.config.suggested_fixed API change, staterror fix, set_poi(None)
```
Comment (alexander-held) [inline setup.cfg:26]:
```suggestion
    pyhf[minuit]~=0.7.0  # model.config.suggested_fixed / .par_names API changes, set_poi(None)
```

**Review by alexander-held — APPROVED** [review]
Looks good, thank you! The pre-commit CI failed once with a strange `mypy` report related to `numpy`, but that must have been some glitch as re-running fixes it and I have no idea what could have broken there anyway.

This PR also undoes the temporary `pyhf` pinning introduced in #366 to make CI pass.

```
* update lower bound of pyhf to v0.7.0
* update model.config.par_names() to model.config.par_names due to pyhf API change
```

**Pull request merged**